scripts/ndjson_to_csv/ndjson_to_csv.py

```diff
@@ -0,0 +1,74 @@
+# /usr/bin/python3
+
+import datetime
+import json
+
+from functools import reduce
+
+from tqdm import tqdm
+import pandas as pd
+
+def flatten(data, parent_key='', sep='.'):
+    items = []
+    for k, v in data.items():
+        new_key = parent_key + sep + k if parent_key else k
+        if isinstance(v, dict):
+            items.extend(flatten(v, new_key, sep).items())
+        elif isinstance(v, list):
+            if not isinstance(v[0], (str, int)):
+                if len(v) > 1:
+                    for index, item in zip(range(len(v)), v):
+                        if index > 0:
+                            items.extend(flatten(
+                                item,
+                                '{}{}'.format(new_key, index+1),
+                                sep
+                            ).items())
+                        else:
+                            items.extend(flatten(item, new_key, sep).items())
+                else:
+                    for item in v:
+                        items.extend(flatten(item, new_key, sep).items())
+            else:
+                list_strings = [str(item) for item in v]
+                items.append((new_key, ' '.join(list_strings)))
+        else:
+            items.append((new_key, v))
+    return dict(items)
+
+
+if __name__=='__main__':
+    print("This tool can convert FHIR data stored in an NDJSON file to a CSV format.")
+    while True:
+        # Ingest the report results
+        filename = input("Enter the name of the NDJSON file you wish to convert: ")
```

Comment:
I would recommend using a CLI interface to grab the file name at run time. Added benefit would also be able to include a `-h` helper blurb for the script itself.

Reply:
Included `-h`, `--help`, and help message. The script takes 2 positional arguments (source + output file name), verifies their existence or non-existence, and then proceeds.